creator-node/src/serviceRegistry.js

```diff
@@ -92,6 +95,22 @@ const initAudiusLibs = async () => {
   return audiusLibs
 }
 
+/**
+ * Get all creator nodes registered on chain. Use a cache to avoid making execessive amounts of chain calls
+ * TODO - make this a job that pulls this data more frequently
+ */
+const writeAllRegisteredCNodesToRedis = async (libs) => {
+  const cacheKey = 'all_registered_cnodes'
+
+  try {
+    let creatorNodes = (await libs.ethContracts.ServiceProviderFactoryClient.getServiceProviderList('creator-node'))
+    creatorNodes = creatorNodes.filter(node => node.endpoint !== config.get('creatorNodeEndpoint'))
+    redisClient.set(cacheKey, JSON.stringify(creatorNodes))
```

Comment:
should this be an expiring key/val that is re-queried on reads? what if the set of cnodes has changed?

Reply:
i thought about this and i used to have it re-query after some time but i changed my mind. basically there's a ciruclar dependency here (described https://github.com/AudiusProject/audius-protocol/pull/895/files/19e0c54e64d75711d038f9c317b2e96131a3cf35#r503017034). no matter where we put this code, unless we refactor rehydrate altogether, there will be a circular dependency. so anything that requires utils could be in jeopardy of having requires break and i didnt want to risk that. so i just made this persist once at the beginning during node startup and figure with weekly or biweekly versions, it should get restarted often enough until we fix the dependency issues